src/Strings.jl

```diff
@@ -1,9 +1,98 @@
 module Strings
 
-export escapehtml, tocameldash, iso8859_1_to_utf8, ascii_lc_isequal
+export HTTPVersion, escapehtml, tocameldash, iso8859_1_to_utf8, ascii_lc_isequal
 
 using ..IOExtras
 
+# A `Base.VersionNumber` is a SemVer spec, whereas a HTTP versions is just 2 digits,
+# This allows us to use a smaller type and more importantly write a simple parse method
+# that avoid allocations.
+"""
+    HTTPVersion(major, minor)
+
+The HTTP version number consists of two digits separated by a
+"." (period or decimal point). The first digit (`major` version)
+indicates the HTTP messaging syntax, whereas the second digit (`minor`
+version) indicates the highest minor version within that major
+version to which the sender is conformant and able to understand for
+future communication.
+
+See [RFC7230 2.6](https://tools.ietf.org/html/rfc7230#section-2.6)
+"""
+struct HTTPVersion
+    major::UInt8
+    minor::UInt8
+end
+
+HTTPVersion(major::Integer) = HTTPVersion(major, 0x00)
+HTTPVersion(v::AbstractString) = parse(HTTPVersion, v)
+HTTPVersion(v::VersionNumber) = convert(HTTPVersion, v)
+# Lossy conversion. We ignore patch/prerelease/build parts even if non-zero/non-empty,
+# because we don't want to add overhead for a case that should never be relevant.
+Base.convert(::Type{HTTPVersion}, v::VersionNumber) = HTTPVersion(v.major, v.minor)
+Base.VersionNumber(v::HTTPVersion) = VersionNumber(v.major, v.minor)
+
+Base.show(io::IO, v::HTTPVersion) = print(io, "HTTPVersion(\"", string(v.major), ".", string(v.minor), "\")")
+
+Base.:(==)(va::VersionNumber, vb::HTTPVersion) = va == VersionNumber(vb)
+Base.:(==)(va::HTTPVersion, vb::VersionNumber) = VersionNumber(va) == vb
+
+Base.isless(va::VersionNumber, vb::HTTPVersion) = isless(va, VersionNumber(vb))
```

Comment:
we either need to define these methods, or else overload `getproperty` such that `r.version` returns a `VersionNumber` and we do all comparisons with `VersionNumbers`. Either seems fine to me.

Reply:
This is not that much code, so this seems fine